src/modules/BootStorage.rb

```diff
@@ -101,29 +103,15 @@ def possible_locations_for_stage1
 
     # Get extended partition for given partition or disk
     def extended_partition_for(device)
-      disk_partition = Yast::Storage.GetDiskPartition(device)
-      return nil unless disk_partition["disk"]
+      staging = Y2Storage::StorageManager.instance.staging
 
-      target_map = Yast::Storage.GetTargetMap
-      disk_map = target_map[disk_partition["disk"]] || {}
-      partitions = disk_map["partitions"] || []
-      ext_part = partitions.find { |p| p["type"] == :extended }
-      return nil unless ext_part
-
-      ext_part["device"]
-    end
-
-    def find_mbr_disk
-      # use the disk with boot partition
-      mp = Storage.GetMountPoints
-      boot_disk = Ops.get_string(
-        mp,
-        ["/boot", 2],
-        Ops.get_string(mp, ["/", 2], "")
-      )
-      log.info "Disk with boot partition: #{boot_disk}, using for MBR"
+      blk_device = Storage::BlkDevice.find_by_name(staging, device)
+      if Storage.partition?(blk_device)
+        partition = Storage.to_partition(blk_device)
+        return partition.parent.name if partition.type == Storage::PartitionType_LOGICAL
+      end
 
```

Comment:
Just pointing a possible alternative using `Refinements::DevicesList`. No offense meant :wink:

```ruby
     def extended_partition_for(device)
        staging = Y2Storage::StorageManager.instance.staging
        logical_partition = staging.partitions.with(name: device).with(type: Storage::PartitionType_LOGICAL).first
        return nil unless logical_partition
        logical_partition.parent.name
    end
```

I don't have a strong preference. The advantage of "my" alternative is that you don't need casting, something that looks weird in Ruby.